clang/lib/CodeGen/CGCall.cpp

```diff
@@ -5031,6 +5031,23 @@ static unsigned getMaxVectorWidth(const llvm::Type *Ty) {
   return MaxVectorWidth;
 }
 
+static bool shouldCreateFPBuiltinForFD(const FunctionDecl *FD, StringRef Name) {
```

Comment:
This is close to what I was thinking, but if getBuiltinID() returns a non-zero value, it still might be a non-fp-related builtin, and if we have an accuracy map, the name might not be in the map. So we could still be calling EmitFPBuiltinofFD() for a lot of functions that don't need it.

On the other hand, I see from even just this limited implementation that there's necessarily going to be duplication of the list of handled functions and builtins between this check and the EmitFPBuiltinofFD() implementation, so maybe it would be better to rename it something like MaybeEmitFPBuiltinofFD() and skip the check I had requested.

This seems like a lot of work for the calls that don't need this, but I don't have an idea to avoid it.

Reply:
Yes, that's the dilemma! There will be duplication. I will remove the check and rename the function unless I come up with something else. But at this point I have looked at it from every corner!